contracts/cw4-stake/src/contract.rs

```diff
@@ -574,6 +630,78 @@ mod tests {
         );
     }
 
+    #[test]
+    fn cw20_token_bond() {
+        let mut deps = mock_dependencies(&[]);
+        cw20_instantiate(deps.as_mut(), Duration::Height(2000));
+
+        // Assert original weights
+        assert_users(deps.as_ref(), None, None, None, None);
+
+        // ensure it rounds down, and respects cut-off
+        bond_cw20(deps.as_mut(), 12_000, 7_500, 4_000, 1);
+
+        // Assert updated weights
+        assert_stake(deps.as_ref(), 12_000, 7_500, 4_000);
+        assert_users(deps.as_ref(), Some(12), Some(7), None, None);
+    }
+
+    #[test]
+    fn cw20_token_claim() {
+        let mut deps = mock_dependencies(&[]);
+        let unbonding = Duration::Height(50);
+        cw20_instantiate(deps.as_mut(), unbonding);
+
+        // bond some tokens
+        bond_cw20(deps.as_mut(), 20_000, 13_500, 500, 1);
+
+        // unbond part
+        unbond(deps.as_mut(), 7_900, 4_600, 0, 10);
+
+        // Assert updated weights
+        assert_stake(deps.as_ref(), 12_100, 8_900, 500);
+        assert_users(deps.as_ref(), Some(12), Some(8), None, None);
+
+        // with proper claims
+        let mut env = mock_env();
+        env.block.height += 10;
```

Comment:
It would be nice to have a `const HEIGHT_DELTA` or so, and use it here and above; just for clarity.